trin-core/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![warn(clippy::unwrap_used)]
+#![warn(clippy::cargo)]
```

Comment:
Any reason this is just in trin-core? Could/should we enable for every module?

Reply:
I was getting some inconsistent behavior tbh. Which is why I only left it in here. However, I was also using `#![warn(unused_crate_dependencies)]` which is a rustc lint. This might have been contributing to the inconsistencies, since I don't believe `rustc` lints and `clippy` lints always play nicely. I could be very wrong about that though. Regardless, yeah both of those lints are very useful and should be enabled everywhere imo - trac